pkg/errors/factory.go

```diff
@@ -0,0 +1,34 @@
+package errors
+
+import (
+	api_errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+func isCRDError(status api_errors.APIStatus) bool {
+	for _, cause := range status.Status().Details.Causes {
+		if strings.HasPrefix(cause.Message, "404")  && cause.Type == v1.CauseTypeUnexpectedServerResponse {
```

Comment:
I inspected the APIStatus error returned by kubernetes and saw that it returned a 404 error within the `causes` array, this does not happen in case it is a usual not found error.